dotnetv3/S3/PutBucketEncryption/ServerSideEncryption/ServerSideEncryption.cs

```diff
@@ -0,0 +1,158 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Net;
+using System.Text.RegularExpressions;
+using Amazon.S3;
+using Amazon.S3.Model;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace ServerSideEncryption;
+
+// snippet-start:[S3.dotnetv3.ServerSideEncryption]
+/// <summary>
+/// Set and read the server side encryption settings for a bucket.
+/// </summary>
+public static class ServerSideEncryption
+{
+    public static IAmazonS3 _s3Client = null!;
+    public static string _bucketName = null!;
+    public static string _kmsKeyId = null!;
+    public static bool _interactive = true;
+
+    static async Task Main(string[] args)
+    {
+        // Use the AWS .NET Core Setup package to set up dependency injection for the Amazon S3 service.
+        // Use your AWS profile name, or leave it blank to use the default profile.
+        using var host = Host.CreateDefaultBuilder(args)
+            .ConfigureServices((_, services) =>
+                services.AddAWSService<IAmazonS3>()
+            ).Build();
+
+        // Now the client is available for injection.
+        _s3Client = host.Services.GetRequiredService<IAmazonS3>();
+
+        Console.WriteLine(new string('-', 80));
+        Console.WriteLine("This example demonstrates setting and getting the server side encryption for a bucket.");
+        Console.WriteLine(new string('-', 80));
+
+        var retries = 5;
+        var success = false;
+        while (!success && retries > 0)
+        {
+            if (_interactive)
+            {
+                _bucketName = PromptUserForResource("Enter the name of a bucket to set encryption:");
+                _kmsKeyId = PromptUserForResource("Enter the Id of the KMS Key to use for encryption:");
+            }
+            success = await SetBucketServerSideEncryption(_bucketName, _kmsKeyId);
+            retries--;
+        }
+
+        if (success)
```

Comment:
Should there be some console output if it's not a success?

Reply:
If the Set operation fails, yes, it will output information in that function. So I didn't repeat that here, it will just go on to "complete".